packages/mui-private-theming/src/ThemeProvider/ThemeProvider.js

```diff
@@ -52,14 +52,19 @@ function ThemeProvider(props) {
   }
 
   const theme = React.useMemo(() => {
-    const output = outerTheme === null ? localTheme : mergeOuterLocalTheme(outerTheme, localTheme);
+    let internalTheme = localTheme;
+    if (identifier && typeof localTheme !== 'function') {
+      internalTheme = localTheme[identifier] || localTheme;
+    }
+    const output =
```

Comment:
This step is a bit confusing to me. Mainly because we are merging the two themes, regardless of whether we use the `identifier`. This means that the theme inside the identifier will have the merged theme from context together with the user-defined theme. As the newly define theme overrides anyway it should be fine, but I am not sure if this is the expected behavior. For e.g. I can access mixins from MD in Joy UI's theme.